Docker can't be build with pnpm@8.1.0

I'm trying to build docker at 1068adc7ff7f6ba8cd22cfc07640e185a50be57e and it fails:

```
$ docker build . --no-cache
[+] Building 3.5s (12/18)                                                                                                                                                 docker:default
 => CACHED [internal] load git source https://github.com/calcom/synclinear.com.git                                                                                                  0.8s
 => resolve image config for docker.io/docker/dockerfile:1.4                                                                                                                        0.4s
 => CACHED docker-image://docker.io/docker/dockerfile:1.4@sha256:9ba7531bd80fb0a858632727cf7a112fbfd19b17e94c4e84ced81e24ef1a0dbc                                                   0.0s 
 => [internal] load metadata for docker.io/library/node:18-alpine                                                                                                                   0.0s
 => [builder 1/9] FROM docker.io/library/node:18-alpine                                                                                                                             0.0s
 => CACHED [builder 2/9] WORKDIR /synclinear                                                                                                                                        0.0s
 => [runtime 3/7] RUN addgroup --system --gid 1001 nodejs                                                                                                                           0.3s
 => [builder 3/9] COPY package.json .                                                                                                                                               0.1s
 => [builder 4/9] RUN corepack enable && corepack prepare                                                                                                                           1.0s
 => [runtime 4/7] RUN adduser --system --uid 1001 nextjs                                                                                                                            0.7s
 => [builder 5/9] COPY pnpm-lock.yaml .                                                                                                                                             0.0s 
 => ERROR [builder 6/9] RUN pnpm fetch                                                                                                                                              1.1s
------                                                                                                                                                                                   
 > [builder 6/9] RUN pnpm fetch:                                                                                                                                                         
1.025  ERR_PNPM_FROZEN_LOCKFILE_WITH_OUTDATED_LOCKFILE  Cannot perform a frozen installation because the version of the lockfile is incompatible with this version of pnpm               
1.025                                                                                                                                                                                    
1.025 Try either:                                                                                                                                                                        
1.025 1. Aligning the version of pnpm that generated the lockfile with the version that installs from it, or
1.025 2. Migrating the lockfile so that it is compatible with the newer version of pnpm, or
1.025 3. Using "pnpm install --no-frozen-lockfile".
1.025 Note that in CI environments, this setting is enabled by default.
------
Dockerfile:13
--------------------
  11 |     
  12 |     COPY pnpm-lock.yaml .
  13 | >>> RUN pnpm fetch
  14 |     COPY . .
  15 |     RUN pnpm install --recursive --offline --frozen-lockfile
--------------------
ERROR: failed to solve: process "/bin/sh -c pnpm fetch" did not complete successfully: exit code: 1
```

But works if I change ` "packageManager": "pnpm@7.33.3"`.

--- a/package.json
+++ b/package.json
@@ -46,6 +46,6 @@
         "react-select": "^5.7.4",
         "uuid": "^9.0.0"
     },
-    "packageManager": "pnpm@8.1.0"
+    "packageManager": "pnpm@7.33.3"
 }
 

